design/dedicated-osd-pod.md

```diff
@@ -23,22 +23,34 @@ A more comprehensive discussions can be found at [this issue](https://github.com
 
 ## Proposal   
 
-We propose the following OSD startup change.
+We propose the following change to address the limitations.
+
+### Terms
+
+- OSD Discovery. A process that is given a device or directory path filter and use the filter to discover devices on the host. The process updates the OSD orchestration Configmap with the device paths and mark the orchestration status as "DeviceDiscovered". The process is created by Rook Operator and runs in a Kubernetes deployment on the storage node. This process is storage backend agnostic. Currently `rook osd` command implements this functionality.
+
+- OSD Provisioner. A process that is given device or directory paths upon start and make backend specific storage types. For now, the process prepare Ceph OSDs. Currently `rook osd` command implements this functionality.
 
 ### Create new OSDs
-| Sequence |Rook Operator  | OSD Provisioner   | Ceph OSD Deployment | 
-|---|---|---|---|
-| 1  | Read devices from cluster CRD and create a Pod for OSD Provisioner for each storage node  |   |   |
-| 2  | Watch Configmap  | Prepare OSDs  |   |
-| 3  | Watch Configmap  |  Persist OSD ID, datapath, and node info in a per node Configmap | |
-| 4  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | |
-| 5  | Create one deployment per OSD | |
-| 6  | | | Start `ceph-osd` Daemon one Pod per device |
+| Sequence |Rook Operator | OSD Discovery  | OSD Provisioner   | Ceph OSD Deployment | 
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create an OSD Discovery deployment on each storage node  |   |   |
+| 2  | Watch Configmap  | Discover OSD devices that fit those defined in Cluster CRD and update Configmap with the device paths| | |
+| 3  | Detect Configmap change, parse Configmap, extract device paths, and create an OSD Provisioner deployment for each device  |Detect device additions or removals | | |
+| 4  | Watch Configmap |Detect device additions or removals | Prepare OSDs, Persist OSD ID, datapath, and node info in a per node Configmap | |
+| 5  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | Detect device additions or removals | | |
+| 6  | Create one deployment per OSD |Detect device additions or removals | |
+| 7  | |Detect device additions or removals | | Start `ceph-osd` Daemon one Pod per device |
```

Comment:
I think step 7 makes sense as detect device additions or removals. But the result should be to return to steps 3-6 to provision and create the osd daemon pod. No other steps would need to "detect device additions or removals"

Reply:
Does this table imply that device discovery is a long running daemon? or does it get launched at certain times?

Reply:
it is a long running process running inside a pod managed by replicaset or deployment